src/integrationTest/groovy/wooga/gradle/unity/tasks/DefaultUnityTaskIntegrationSpec.groovy

```diff
@@ -51,4 +57,75 @@ class DefaultUnityTaskIntegrationSpec extends UnityTaskIntegrationSpec<Unity> {
         fileExists(project_path, "Library")
         fileExists(project_path, "ProjectSettings")
     }
+
+    // TODO: Should these be moved onto every task?
+    @Unroll
+    def "can set command line argument option #option when #location with key #keyString"() {
+
+        when:
+        def query = runPropertyQuery(getter, setter)
+
+        then:
+        query.success
+        query.matches(value, type)
+
+        where:
+        [testCase, value, location] <<
+            [
+                UnityCommandLineOption.argumentFlags
+                    .findAll({ it -> it.map })
+                    .collect({ it -> [it.toString()] }),
+                ["foobar"],
+                [PropertyLocation.property, PropertyLocation.environment]
+            ].combinations()
+
+        option = testCase[0]
+        type = String.class
+        propertyKey = "unity." + option
+        envKey = PropertyUtils.envNameFromProperty(propertyKey)
+        keyString = location == PropertyLocation.property
+            ? propertyKey : envKey
+
+        setter = new PropertySetterWriter(subjectUnderTestName, option)
+            .set(value, type)
+            .withPropertyKey(propertyKey)
+            .withEnvironmentKey(envKey)
+            .to(location)
+        getter = new PropertyGetterTaskWriter(setter)
+    }
+
+    @Unroll
+    def "can set command line flag option #option to #value when #location with key #keyString"() {
+
+        when:
+        def query = runPropertyQuery(getter, setter)
+
+        then:
+        query.success
+        query.matches(value, type)
+
+        where:
+        [testCase, value, location] <<
+            [
+                UnityCommandLineOption.flags
+                    .findAll({ it -> it.map })
+                    .collect({ it -> [it.toString()] }),
```

Comment:
This is a false prophesy. You create tests based on the list provided by the implementation.
You want to veryfy that the code provides a specific list of flags etc. And that must be static otherwise you won't catch regressions or breaking changes when removing/renaming flags.

Just to make this clear. If the code provides a list of 3 flags:  `Foo`, `Bar` and `Baz` then this test would check if all mappigns etc are setup which is great. But if a code chagne would remove one of the flags this test would not find this out. Because the single point of truth is the code. But the test spec is the single point of truth. Thats why we can't pin the tests to the implementation. It's like if a Policeman asks a robber if he commited a crime or not.

Reply:
I understand; I forgot to write a 3rd test where these are done by hand to pin them.